src/Entity/SmsDeliveryReport.php

```diff
@@ -89,30 +96,64 @@ public function setStatusMessage($message) {
   /**
    * {@inheritdoc}
    */
+  public function getStatusTime() {
+    return $this->get('status_time')->value;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setStatusTime($timestamp) {
+    $this->set('status_time', $timestamp);
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
   public function getTimeQueued() {
-    return $this->get('time_queued')->value;
+    $queued = $this->getRevisionAtStatus(SmsMessageReportStatus::QUEUED);
+    return $queued ? $queued->getStatusTime() : NULL;
   }
 
   /**
    * {@inheritdoc}
    */
   public function setTimeQueued($time) {
```

Comment:
Thats ok. The system is revisions, there can be multiple of the same status over time. The rule is: only one status at a time.